client/src/sync-filesystem.js

```
@@ -75,6 +252,27 @@ function SyncFileSystem(fs) {
       // second for some.
       var pathOrFD = args[pathArgPos];
 
+      function wrappedCallback() {
+        var args = Array.prototype.slice.call(arguments, 0);
+        if(args[0] || mode === syncModes.DELETE) {
+          return callback.apply(null, args);
+        }
+
+        setUnsyncedFn(pathOrFD, function(err) {
+          if(err) {
+            return callback(err);
+          }
+          callback.apply(null, args);
+        });
+      }
+
+      args[lastIdx] = wrappedCallback;
+
+      if(mode === syncModes.DELETE && pathOrFD === root) {
+        // Deal with deletion of the sync root
+        // https://github.com/mozilla/makedrive/issues/465
```

Let's at least warn here, so we don't silently pass this over.